plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/NonSystemJobSuspender.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.core.runtime.jobs.Job;
+
+/**
+ * Prevents scheduling all future non-system jobs once {@link #suspendFutureJobs} is called, until
+ * {@link #resume} is called. Jobs already scheduled are not affected and will run to completion.
+ *
+ * The class is for https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1155. The
+ * purpose of the class is to prevent the second ConvertJob from running. The second ConvertJob is
+ * triggered by the first ConvertJob when the latter job installs the JSDT facet.
+ *
+ * Not recommended to use for other situations, although the workings of the class are general.
+ */
+public class NonSystemJobSuspender {
```

Comment:
Every class loader (and there are many in OSGI) that sees this load a new instance of this class. Does this matter to us? Or is only a single class loader ever going to see this? 

Can we make this class non-public (package-protected). That might be a little safer.